src/QMCHamiltonians/tests/test_ion_derivs.cpp

```diff
@@ -772,4 +772,778 @@ TEST_CASE("Eloc_Derivatives:multislater_wj", "[hamiltonian]")
+
+  std::vector<std::vector<ValueMatrix_t>> dM; //Derivative of slater matrix.
+  std::vector<std::vector<ValueMatrix_t>> dB; //Derivative of B matrices. 
+  matlist.push_back(upmat);
+  matlist.push_back(dnmat);
+
+  dM.push_back(matlist);
+  dM.push_back(matlist);
+  dM.push_back(matlist);
+
+  dB.push_back(matlist);
+  dB.push_back(matlist);
+  dB.push_back(matlist);
+
+  B.push_back(upmat);
+  B.push_back(dnmat);
+
+  X.push_back(upmat);
+  X.push_back(dnmat);
+
+//  twf.get_M(elec, matlist);
+
+  OperatorBase* kinop = ham->getHamiltonian(KINETIC);
+
+//  kinop->evaluateOneBodyOpMatrix(elec, twf, B);
+
+  
+  std::vector<ValueMatrix_t> minv;
+  std::vector<ValueMatrix_t> B_gs, M_gs; //We are creating B and M matrices for assumed ground-state occupations. 
+                                         //These are N_s x N_s square matrices (N_s is number of particles for species s).
+  B_gs.push_back(upmat);
+  B_gs.push_back(dnmat);
+  M_gs.push_back(upmat);
+  M_gs.push_back(dnmat);
+  minv.push_back(upmat);
+  minv.push_back(dnmat);
+
+
+//  twf.get_M(elec, matlist);
+  std::vector<std::vector<ValueMatrix_t>> dB_gs;
+  std::vector<std::vector<ValueMatrix_t>> dM_gs;
+  std::vector<ValueMatrix_t> tmp_gs;
+//  twf.get_gs_matrix(B, B_gs);
+//  twf.get_gs_matrix(matlist, M_gs);
+//  twf.invert_M(M_gs, minv);
+//  twf.build_X(minv, B_gs, X);
+  for (int id = 0; id < matlist.size(); id++)
+  {
+//    int ptclnum = twf.num_particles(id);
+    int ptclnum = (id==0 ? Nup : Ndn); //hard coded until twf interface comes online.  
+    ValueMatrix_t gs_m;
+    gs_m.resize(ptclnum, ptclnum);
+    tmp_gs.push_back(gs_m);
+  }
+
+
+  dB_gs.push_back(tmp_gs);
+  dB_gs.push_back(tmp_gs);
+  dB_gs.push_back(tmp_gs);
+
+  dM_gs.push_back(tmp_gs);
+  dM_gs.push_back(tmp_gs);
+  dM_gs.push_back(tmp_gs);
+
+  //Finally, we have all the data structures with the right dimensions.  Continue.
+
+  ParticleSet::ParticleGradient_t fkin_complex(ions.getTotalNum());
+  ParticleSet::ParticlePos_t fkin(ions.getTotalNum());
+
+
+  for (int ionid = 0; ionid < ions.getTotalNum(); ionid++)
+  {
+    for (int idim = 0; idim < OHMMS_DIM; idim++)
+    {
+//      twf.wipe_matrix(dB[idim]);
+//      twf.wipe_matrix(dM[idim]);
+    }
+
+//    twf.get_igrad_M(elec, ions, ionid, dM);
+//    kinop->evaluateOneBodyOpMatrixForceDeriv(elec, ions, twf, ionid, dB);
+
+    for (int idim = 0; idim < OHMMS_DIM; idim++)
+    {
+//      twf.get_gs_matrix(dB[idim], dB_gs[idim]);
+//      twf.get_gs_matrix(dM[idim], dM_gs[idim]);
+//      fkin_complex[ionid][idim] = twf.compute_gs_derivative(minv, X, dM_gs[idim], dB_gs[idim]);
+    }
+    convertToReal(fkin_complex[ionid], fkin[ionid]);
+  }
+
+
+  ValueType keval = 0.0;
+  RealType keobs  = 0.0;
+//  keval           = twf.trAB(minv, B_gs);
+  convertToReal(keval, keobs);
+//  REQUIRE(keobs == Approx(9.1821937928e+00));
+#if defined(MIXED_PRECISION)
+//  REQUIRE(fkin[0][0] == Approx(1.0852823603357820).epsilon(1e-4));
```

Comment:
In practice if the test is cheap its better to use CHECK to continue and see if all elements are off.

Reply:
Done.  I kept REQUIRE's on return values that should absolutely work as a precondition for the derivative evaluations, and on major conceptual points (e.g. returned derivative vector should be nonzero).  Extensive tests on numerics have been changed to CHECK.

Reply:
Yes. At the places that the code is pointless to continue, use `REQUIRE`